### PR TITLE
Som devel 20 wall

### PR DESCRIPTION
**Pull request by sorgom**

### CODE REVIEW

File: application/components/TSW/src/TSW_Hub.cpp [hunk 1]
```diff
@@ -29,18 +29,14 @@ void TSW_Hub::fromDsp(const size_t pos, const GuiCmd& tele) const
 
 void TSW_Hub::toFld(const size_t id, const INT32 cmd) const
 {
-    static CmdFld tele;
-    Mem::zero(tele);
-    tele.cmd1 = cmd;
-    IL::getDispatcher().dispatch(id, tele);
+    // CmdFld tele(cmd);
```
Comment (sombeam) [inline application/components/TSW/src/TSW_Hub.cpp]: remove out-commented lines

File: application/components/TSW/src/TSW_Hub.cpp [hunk 2]
```diff
@@ -29,18 +29,14 @@ void TSW_Hub::fromDsp(const size_t pos, const GuiCmd& tele) const
...
+    IL::getDispatcher().dispatch(id, CmdFld(cmd));
 }
 
 void TSW_Hub::toGui(const size_t id, const INT32 state) const
 {
-    static StateGui tele;
-    Mem::zero(tele);
-    tele.state1 = state;
-    IL::getDispatcher().dispatch(id, tele);
+    // StateGui tele(state);
```
Comment (sombeam) [inline application/components/TSW/src/TSW_Hub.cpp]: remove out-commented lines

File: testing/testenv/testlib/GenProjData.h [hunk 3]
```diff
@@ -82,11 +99,28 @@ namespace test
             setType(mLCRs, type);
         }
 
+        bool dump(CONST_C_STRING filename) const
```
Comment (sombeam) [inline testing/testenv/testlib/GenProjData.h]: does anybody evaluate the return?
make it void!

File: testing/testenv/testlib/MemStream.h
Comment (sombeam) [inline testing/testenv/testlib/MemStream.h:1]: nice feature - but not used anywhere
remove to some other place!

File: testing/tests/devtests/DT_02.cpp [hunk 4]
```diff
@@ -1,10 +1,12 @@
 //  ============================================================
 //  test of test features
 //  - output of fixed size strings
+//  - MemStream
```
Comment (sombeam) [inline testing/tests/devtests/DT_02.cpp:4]: remove

File: testing/tests/devtests/DT_02.cpp [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
 //  ============================================================
 //  created by Manfred Sorgo
 
 #include <testlib/TestGroupBase.h>
+#include <testlib/MemStream.h>
```
Comment (sombeam) [inline testing/tests/devtests/DT_02.cpp]: remove

File: testing/tests/devtests/DT_02.cpp [hunk 6]
```diff
@@ -17,7 +19,9 @@ namespace test
     TEST(DT_02, T01)
     {
         STEP(1)
-        ElementName fn = {{ '0', '1', '2', '3', '4', '5', '6', '7', '8' }};
+        const CHAR chars[9] = { '0', '1', '2', '3', '4', '5', '6', '7', '8' };
```
Comment (sombeam) [inline testing/tests/devtests/DT_02.cpp]: avoid magic numbers

File: testing/tests/devtests/DT_02.cpp [hunk 7]
```diff
@@ -31,4 +35,26 @@ namespace test
         STRCMP_EQUAL("0123*56*8", os.str().c_str());
     }
 
+    TEST(DT_02, T02)
```
Comment (sombeam) [inline testing/tests/devtests/DT_02.cpp]: MemStream not used.
remove test to some other place

File: testing/tests/devtests/DT_03.cpp [hunk 8]
```diff
@@ -20,7 +20,6 @@ namespace test
     {
         SETUP()
         GenProjData<CAPTSW, CAPSIG, CAPLCR, CAPSEG> projData;
-        // projData.preset();
```
Comment (sombeam) [inline testing/tests/devtests/DT_03.cpp:23]: remove out-commented

File: testing/tests/moduletests/BAS/BAS_01.cpp [hunk 9]
```diff
@@ -4,6 +4,7 @@
 //  created by Manfred Sorgo
 
 #include <testlib/TestGroupBase.h>
+#include <testlib/MemStream.h>
```
Comment (sombeam) [inline testing/tests/moduletests/BAS/BAS_01.cpp]: remove

File: testing/tests/moduletests/LCR/LCR_02.cpp [hunk 10]
```diff
@@ -83,10 +76,7 @@ namespace test
...
     {
         SETUP()
-        StateGui tele;
-        Mem::zero(tele);
-        tele.state1 = 401;
-        tele.state2 = 402;
+        StateGui tele(401, 402);
         
         STEP(1)
         m_Dispatcher().expectDispatch(4, tele);
```
Comment (sombeam) [inline testing/tests/moduletests/LCR/LCR_02.cpp]: would it be possible to make expectations by rvalue?
please check!

File: testing/tests/moduletests/SYS/SYS_02.cpp [hunk 11]
```diff
@@ -152,36 +148,34 @@ namespace test
         SETUP()
         StateGui stateIn;
```
Comment (sombeam) [inline testing/tests/moduletests/SYS/SYS_02.cpp]: redundant?

File: testing/tests/moduletests/SYS/SYS_02.cpp [hunk 12]
```diff
@@ -112,36 +110,34 @@ namespace test
         SETUP()
         CmdFld cmdIn;
```
Comment (sombeam) [inline testing/tests/moduletests/SYS/SYS_02.cpp]: remove?

**Review by sombeam — CHANGES_REQUESTED** [review]

File: application/components/TSW/src/TSW_Hub.cpp (hunk 1 above)
Comment (sorgom) [reply application/components/TSW/src/TSW_Hub.cpp]: done

File: application/components/TSW/src/TSW_Hub.cpp (hunk 2 above)
Comment (sorgom) [reply application/components/TSW/src/TSW_Hub.cpp]: done

File: testing/testenv/testlib/GenProjData.h (hunk 3 above)
Comment (sorgom) [reply testing/testenv/testlib/GenProjData.h]: made it void

File: testing/tests/moduletests/BAS/BAS_01.cpp (hunk 9 above)
Comment (sorgom) [reply testing/tests/moduletests/BAS/BAS_01.cpp]: done

File: testing/tests/moduletests/SYS/SYS_02.cpp (hunk 11 above)
Comment (sorgom) [reply testing/tests/moduletests/SYS/SYS_02.cpp]: yes, removed

File: testing/tests/moduletests/SYS/SYS_02.cpp (hunk 12 above)
Comment (sorgom) [reply testing/tests/moduletests/SYS/SYS_02.cpp]: yes, removed

File: testing/tests/moduletests/LCR/LCR_02.cpp (hunk 10 above)
Comment (sorgom) [reply testing/tests/moduletests/LCR/LCR_02.cpp]: yes, implemented

**Review by sombeam — DISMISSED** [review]

File: testing/tests/devtests/DT_02.cpp (hunk 6 above)
Comment (sorgom) [reply testing/tests/devtests/DT_02.cpp]: -> NumElementNameChars

**Review by sombeam — APPROVED** [review]

**Pull request merged**